### PR TITLE
Enabling width scaling experiments for ResNets to study alpha values with change in base width

### LINKED ISSUE #34: Enable varying base width of Resnet18 models for running sweeps
- [x] Add ResNet18 model definition in backbone file
- [x] Add base model width parameter to control the base width
- [x] Infer base width from model name
- [x] [Optional] Add argument for model base width
- [x] Incorporate new parameter in barlowtwins and linear models (Only for testing new argument option for base width)
- [x] Change log directory name to add width in folder structure
- [x] Run coarse BarlowTwins sweep to check good hyperparams
- [x] Run sweep across different widths

New todos:
- [x] Remove the base model width argument and simply infer width of ResNet18 from model name
- [x] Add train accuracy along with test accuracy at each epoch
- [x] Save precached features to logdir
- [x] Re-run sweep over widths
- [x] Incorporate wandb to log results to server
- [ ] Add width variation for ResNet50

### PR DESCRIPTION
**Pull request by arnaghosh**
This PR should close #34. 
**TL;DR: Allows passing base width value in model name and updates the model architecture to use this info.**

**Summary of changes:**
1. [models/backbone.py](fastssl/models/backbone.py): Adding Resnet18 & ResNet50 architecture and enabling setting base width from model name.
2. [models/resnet.py](fastssl/models/resnets.py): New script containing definition of different Resnet configs.
3. [utils/base.py](fastssl/utils/base.py): Added util functions to initialize wandb with group and project name, as well as to log results as a dictionary.
4. [sbatch_sh_files/resnet18_barlow_sweep.sh](sbatch_sh_files/resnet18_barlow_sweep.sh): The bash script file that was used to run hyperparameter sweep for original ResNet18 model.
5. [sbatch_sh_files/resnet18_width_barlow.sh](sbatch_sh_files/resnet18_width_barlow.sh): The bash script file that was used to run sweep for ResNet18 base width.
6. [sbatch_sh_files/resnet50_width_barlow.sh](sbatch_sh_files/resnet50_width_barlow.sh): The bash script file that was used to run sweep for ResNet50 base width.
7. [scripts/train_model_widthVary.py](scripts/train_model_widthVary.py): Slight variation from original entry point `train_model.py` that was used to run the width vary experiments. This contains the wandb logging information.

**Results:**
Sweep results are logged on wandb. Check no label noise linear eval [results](https://wandb.ai/eigengroup/modelWidth-scaling) and 15% label noise [results](https://wandb.ai/eigengroup/modelWidth-scaling_Noise15).

### CODE REVIEW

File: fastssl/models/resnets.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    def _make_layer(self, block, planes, num_blocks, stride):
+        strides = [stride] + [1]*(num_blocks-1)
+        layers = []
+        for stride in strides:
+            layers.append(block(self.in_planes, planes, stride, no_bn=self.no_bn))
+            self.in_planes = planes * block.expansion
+        return nn.Sequential(*layers)
+
+    def forward(self, x):
+        out = F.relu(self.bn1(self.conv1(x)))
```
Comment (kumarkrishna) [inline fastssl/models/resnets.py:88]: Unless we want to log the intermediate features, I'd suggest putting all layers in nn.Sequential.

**Review by kumarkrishna — COMMENTED** [review]
LGTM with minor comments. Skimmed through the scripts

File: fastssl/models/resnets.py (hunk 1 above)
Comment (arnaghosh) [reply fastssl/models/resnets.py:88]: All layers of Resnet? I thought that maybe we could keep them separate in case we want to investigate individual blocks in the future. Having them separate might come in handy to attach forward hooks.

**Review by kumarkrishna — APPROVED** [review]
